source_code/main_mcu/src/LOGIC/logic_database.c

```diff
@@ -604,7 +614,8 @@ RET_TYPE logic_database_add_webauthn_credential_for_service(uint16_t service_add
     memset((void*)&temp_cnode, 0, sizeof(temp_cnode));
     
     /* Copy everything */
-    memcpy(temp_cnode.user_handle, user_handle, sizeof(temp_cnode.user_handle));
+    temp_cnode.user_handle_len = user_handle_len;
+    memcpy(temp_cnode.user_handle, user_handle, user_handle_len);
```

Comment:
please sanitize user_handle_len or return error if the value is higher than the field itself